indexing.sql

```diff
@@ -0,0 +1,84 @@
+DROP TABLE IF EXISTS car_models;
+
+\i ./scripts/car_models.sql;
+
+\i ./scripts/car_model_data.sql;
+\i ./scripts/car_model_data.sql;
+\i ./scripts/car_model_data.sql;
+\i ./scripts/car_model_data.sql;
+\i ./scripts/car_model_data.sql;
+\i ./scripts/car_model_data.sql;
+\i ./scripts/car_model_data.sql;
+\i ./scripts/car_model_data.sql;
+\i ./scripts/car_model_data.sql;
+\i ./scripts/car_model_data.sql;
+
+-- 1: 22.018 ms
+SELECT DISTINCT make_title
+FROM car_models
+WHERE make_code='LAM';
+
+-- 2: 16.753 ms
+SELECT DISTINCT model_title
+FROM car_models
+WHERE make_code='NISSAN'
+    AND model_code='GT-R';
+
+-- 3: 23.122 ms
+SELECT make_code, model_code, model_title, year
+FROM car_models
+WHERE make_code='LAM';
+
+-- 4: 250.579 ms
+SELECT *
+FROM car_models
+WHERE year
+BETWEEN 2010
+AND 2015;
+
+-- 5: 68.370 ms
+SELECT *
+FROM car_models
+WHERE year=2010;
+
+-- INDEXING
+
+CREATE INDEX idx
+ON car_models(make_code, make_title, year);
```

Comment:
So you should only make an index on make_code because your where clauses only really uses that one. You could also create a index on make_code and model_code since one of your where clauses uses it but it doesn't give you that much of a faster result. Year does not see the same benefits because for each year, there's still a lot of data, so even after sorting them by year, you'd still have to do a lot of searching.